From 591bb047cbb0cb29841fde39964cbd78fb2bff6c Mon Sep 17 00:00:00 2001
From: Jerry Hoemann <jerry.hoemann@hpe.com>
Date: Tue, 20 May 2025 09:38:57 +0200
Subject: [PATCH 30/45] dmioem: Add type HPE Gen 12

Recognize Gen 12 ProLiant systems.

Signed-off-by: Jerry Hoemann <jerry.hoemann@hpe.com>
Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 dmioem.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dmioem.c b/dmioem.c
index 4a1f3f7..ffbcd95 100644
--- a/dmioem.c
+++ b/dmioem.c
@@ -303,11 +303,12 @@ static void dmi_print_hp_net_iface_rec(u8 id, u8 bus, u8 dev, const u8 *mac)
 	}
 }
 
-typedef enum { G6 = 6, G7, G8, G9, G10, G10P, G11 } dmi_hpegen_t;
+typedef enum { G6 = 6, G7, G8, G9, G10, G10P, G11, G12 } dmi_hpegen_t;
 
 static int dmi_hpegen(const char *s)
 {
 	struct { const char *name; dmi_hpegen_t gen; } table[] = {
+		{ "Gen12",	G12 },
 		{ "Gen11",	G11 },
 		{ "Gen10 Plus",	G10P },
 		{ "Gen10",	G10 },
-- 
2.47.0

